tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -820,7 +869,11 @@ TreeMap<String, Entry<Optional<GroupState>, Optional<Collection<MemberAssignment
                         consumer.clientId(),
                         consumer.groupInstanceId().orElse(""),
                         consumer.assignment().topicPartitions().size(),
-                        new ArrayList<>(verbose ? consumer.assignment().topicPartitions() : Collections.emptySet())
+                        new ArrayList<>(verbose ? consumer.assignment().topicPartitions() : Collections.emptySet()),
+                        consumer.targetAssignment().isEmpty() ? List.of() : new ArrayList<>(consumer.targetAssignment().get().topicPartitions()),
```

Comment:
nit: You may be able to write it as follow: `consumer.targetAssignment().map(a -> a.topicPartitions().stream().collect(Collectors.toUnmodifiableList())).orElse(Collections.emptyList()),`.